libkbfs/keybase_service_base.go

```diff
@@ -741,6 +772,22 @@ func (k *KeybaseServiceBase) FSSyncStatusRequest(ctx context.Context,
 	return k.kbfsClient.FSSyncStatus(ctx, resp)
 }
 
+// TeamChanged implements keybase1.NotifyTeamInterface for
+// KeybaseServiceBase.
+func (k *KeybaseServiceBase) TeamChanged(
+	ctx context.Context, arg keybase1.TeamChangedArg) error {
+	k.log.CDebugf(ctx, "Flushing cache for team %s/%s "+
+		"(membershipChange=%t, keyRotated=%t, renamed=%t)",
+		arg.TeamName, arg.TeamID, arg.Changes.MembershipChanged,
+		arg.Changes.KeyRotated, arg.Changes.Renamed)
+	k.setCachedTeamInfo(arg.TeamID, TeamInfo{})
+
+	if arg.Changes.Renamed {
```

Comment:
`TeamChanged` is how we learn of role upgrades, yes.  But so far I haven't done anything to plumb that down to the journal level.  The mdserver will start rejecting the writes of a non-writer being put by the journal, and the journal will see that error and start backing off its attempts.  Eventually if the reader is upgraded, the journal writes will succeed again.  I'm not sure we want it to even permanently stop.

If someone else writes to the folder while we're in this state, we should be able to run CR locally and "rebase" the outstanding changes onto the TLF, so the user will see both the local and remote updates, but we'll be unable to push the changes.  I haven't tested this though.